### PR TITLE
add test for pubspec parsing, and add transformer config to it

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: lib/src/bazelify/pubspec.dart [hunk 1]
```diff
@@ -23,7 +23,25 @@ class Pubspec {
   final Map _pubspecContents;
 
   /// Create a [Pubspec] by parsing [pubspecYaml].
-  Pubspec.parse(String pubspecYaml) : _pubspecContents = loadYaml(pubspecYaml);
+  Pubspec.parse(String pubspecYaml) : _pubspecContents = loadYaml(pubspecYaml) {
+    _transformers = [];
```
Comment (natebosch) [inline lib/src/bazelify/pubspec.dart]: should this be initialized inline?

File: lib/src/bazelify/pubspec.dart [hunk 2]
```diff
@@ -67,4 +85,22 @@ class Pubspec {
 
   /// Name of the package.
   String get pubPackageName => _pubspecContents['name'];
+
+  List<Transformer> _transformers;
```
Comment (natebosch) [inline lib/src/bazelify/pubspec.dart]: we've typically put fields near the top of the class. Could we do that here?

**Review by natebosch — APPROVED** [review]

File: lib/src/bazelify/pubspec.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/bazelify/pubspec.dart]: ya good call, that lets me make it final as well

File: lib/src/bazelify/pubspec.dart (hunk 2 above)
Comment (jakemac53) [inline lib/src/bazelify/pubspec.dart]: Sure, I would actually move all the getters above as well but I don't think there is any consensus around that really so I won't touch things.

**Pull request merged**